### PR TITLE
Bug 1287501 - Fix to last commit where last_modified param may get re…

### PR DESCRIPTION
**Pull request by camd**
…moved entirely

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1704)

### CODE REVIEW

**Comment by edmorley** [comment]
I've just spotted there are people using other last_modified terms aside from greater than, eg both `last_modified__lte` and `last_modified__gte` in:
https://rpm.newrelic.com/accounts/677903/applications/4180461/traced_errors/8b4f5016-4d41-11e6-9e2c-c81f66b8ceca_27674_32227

Whether that's a use-case we should continue to support is another matter (and deploying this hotfix in the meantime is something we should do regardless).

**Comment by edmorley** [comment]
Could we just remove the `pop()` from `UrlQueryFilter.get()` instead?

A quick glance at the uses of `UrlQueryFilter` show that most times people are calling `.pop()` directly, so changing `get()` shouldn't effect much I think?

**Comment by edmorley** [comment]
We should also add some tests for the last_modified behaviour - if it breaks we basically cause Treeherder to stop working since it maxes out the DB CPU, so it probably should have test coverage :-)

**Comment by camd** [comment]
I took out the filter pop footgun.  And added some tests, though pretty simple.  I thought we might want to discuss the policy for these urls going forward.  I think we are vulnerable all over the place.  But this at least guarantees that the last_modified is a valid date.

File: treeherder/webapp/api/jobs.py [hunk 1]
```diff
@@ -94,27 +92,23 @@ def list(self, request, project, jm):
...
+                datestr = lm[1]
+                try:
+                    # ensure last_modified is a date
+                    parser.parse(datestr)
+                except ValueError:
                     return Response(
-                        "`last_modified` of {} is not within last {} minutes".format(
-                            datestr,
-                            LAST_MODIFIED_WINDOW),
+                        "Invalid date value for `last_modified`: ".format(datestr),
```
Comment (edmorley) [inline treeherder/webapp/api/jobs.py]: The `{}` is missing

File: tests/webapp/api/test_jobs_api.py [hunk 2]
```diff
@@ -388,3 +389,16 @@ def test_job_create(webapp, test_repository, test_user, eleven_job_blobs, monkey
...
+@pytest.mark.parametrize('lm_value,expected', [
+    ("2016-07-18T22:16:58.000", 200),
+    ("-Infinity", HTTP_400_BAD_REQUEST),
+    ("whatever", HTTP_400_BAD_REQUEST),
+    ])
+def test_last_modified(webapp, test_project, lm_value, expected):
+    url = reverse("jobs-list", kwargs={"project": test_project})
+    final_url = url + ("?last_modified__gt={}&count=5".format(lm_value))
+
+    resp = webapp.get(final_url, expect_errors=(expected != 200))
```
Comment (edmorley) [inline tests/webapp/api/test_jobs_api.py]: I meant more we also need a test to ensure we're actually filtering by date_modified correctly (ie that we haven't dropped the condition).

**Comment by edmorley** [comment]
Could you also make the first line of the commit message be < 72 chars so it doesn't get wrapped, and add some more details in the body explaining the use-case we're wanting to continue to support?

**Pull request merged**